File: Dataset/Leetcode/valid/55/435.py
 class Solution:
    def XXX(self, nums: List[int]) -> bool:
        if (len(nums) == 0): return False
        if (len(nums) == 1): return True
        i = len(nums) - 2
        last = len(nums) - 1
        while (i >= 0):
            if (nums[i] >= last - i): last = i
            i = i - 1
        return last == 0

File: Dataset/Leetcode/train/75/435.cpp
 class Solution {
public:
    void XXX(vector<int>& nums) {
        int n = nums.size(), zero_r = -1, two_l = n;
        for (int i = 0; i < two_l; ++i) {
            while (nums[i] != 1 && zero_r < i && i < two_l) {
                if (nums[i] == 0) {
                    swap(nums[i], nums[++zero_r]);
                } else {
                    swap(nums[i], nums[--two_l]);
                }
            }
        } 
    }
};

